README.md

```diff
@@ -1 +1,226 @@
+
+Don't forget to modify the `webpack.config.js` to set your wanted export file name.
+
+We provide a `changelog.json` to explain all change in the template, you can remove it and have your own changelog file.
+
+## Name your plugin
+
+By default, in this template we use "MyPlugin" as your plugin name.
+
+So once your plugin's name is choose, you have to make these modifications:
+
+* Rename these files and their content:
+  * `src/draw/FruitDrawer.js`
+  * `src/metadata/FruitMetadata.js`
+  * `src/parser/FruitParser.js`
+  * `src/render/FruitRenderer.js`
+* Change the content of `src/index.js`
+
+Example: You want to name your plugin `Fruit`
+
+`src/draw/FruitDrawer.js` becomes `src/draw/FruitDrawer.js`
+```js
+import { DefaultDrawer } from 'leto-modelizer-plugin-core';
+
+class FruitDrawer extends DefaultDrawer {}
+
+export default FruitDrawer;
+```
+
+Make the same thing for `MyPluginMetadata`, `MyPluginParse` and `MyPluginRenderer`.
+
+Content of `src/index.js` become:
+```js
+import FruitDrawer from './draw/FruitDrawer';
+import FruitMetadata from './metadata/FruitMetadata';
+import FruitParser from './parser/FruitParser';
+import FruitRenderer from './render/FruitRenderer';
+import resources from './assets';
+
+export default {
+  PluginDrawer: FruitDrawer,
+  PluginMetadata: FruitMetadata,
+  PluginParser: FruitParser,
+  PluginRenderer: FruitRenderer,
+  resources,
+};
+```
+
+## Write your metadata
+
+To make your plugin works we need to have an array of [ComponentDefinitions](https://github.com/ditrit/leto-modelizer-plugin-core/blob/main/src/models/ComponentDefinition.js).
+
+The [DefaultMetadata](https://github.com/ditrit/leto-modelizer-plugin-core/blob/main/src/metadata/DefaultMetadata.js) provides two methods:
+* `validate` to validate the metadata
+* `getComponentDefinitions` to get all ComponentDefinitions from your provided metadata
+
+To keep our previous example, we can Have 4 ComponentDefinitions for our `FruitPlugin`:
+- `Bag`, it's a bag that can contains fruits. It has only one mandatory attribute, it's `maxSize` that represent the maximum of fruit it can care. 
+- `Fruit`, it's a simple fruit. Its only mandatory attribute is `name`.
+
+in this example, we don't want to parse and validate a metadata file.
+
+But you can find an implementation example [here](https://github.com/ditrit/iactor/tree/dev/src/metadata) which can validate and get definitions from [this metadata](https://github.com/ditrit/iactor/blob/dev/tests/metadata/aws.json).  
+
+So, here is our `FruitMetadata` class: 
+```js
+import {
+  DefaultMetadata,
+  ComponentDefinition,
+} from 'leto-modelizer-plugin-core';
+
+class FruitMetadata extends DefaultMetadata {
+  getComponentDefinitions() {
+    const bagDefinition = new ComponentDefinition('bag');
+    bagDefinition.definedAttributes.push(new ComponentAttributeDefinition(
+      'maxSize', // Attribute name
+      'Number',  // Attribute type
+      [],
+      true, // is required
+    ));
```

Comment:
Maybe the `ComponentAttributeDefinition` constructor should use an object as argument to get named parameters ?

Reply:
I'm still thinking about this, but yeah we have to improve it :)